Tests/Unit/Components/WebService/Rest/RestConfigurableProducerTest.php

```diff
@@ -115,6 +134,99 @@ public function testItShouldCreateProperRestExceptions(
+                'uri' => 'v0/put/cats',
+                'overRideBaseUri' => 'http://dev.machine/api/',
+                'expectedHost' => 'dev.machine',
+                'expectedPath' => '/api/v0/put/cats',
+            ],
+            'override but ignore null' => [
+                'base_uri' => 'http://someservice.com/api/',
+                'uri' => 'v0/put/cats',
+                'overRideBaseUri' => null,
+                'expectedHost' => 'someservice.com',
+                'expectedPath' => '/api/v0/put/cats',
+            ],
+        ];
+    }
+
+    /**
+     * @param $baseUri
+     * @param $requestUri
+     * @param $overRideBaseUri
+     * @param $expectedHost
+     * @param $expectedPath
+     * @dataProvider getCasesForBaseUri
+     */
+    public function testCanOverrideBaseUrI($baseUri, $requestUri, $overRideBaseUri, $expectedHost, $expectedPath)
+    {
+        $mockHandler = new MockHandler([
+            new Response(200, ['X-Foo' => 'Bar']),
+        ]);
+
+        $container = [];
+        $history = Middleware::history($container);
+        $stack = HandlerStack::create($mockHandler);
+        $stack->push($history);
+
+        $client = new Client(['handler' => $stack]);
+
+        $producer = new RestConfigurableProducer();
+        $producer->setHttpClient($client);
+        $producer->setEvaluator($this->evaluator);
+        $producer->setSerializer($this->serializer);
+        $producer->setConfHelper($this->helper);
+        $producer->setConfHelper($this->helper);
+        $producer->setEventDispatcher($this->eventDispatcher);
+
+        $actionParams = [
+            RestConfigurableProducer::REQUEST_NAME => 'someRequest',
+            RestConfigurableProducer::REQUEST_HTTP_VERB => 'POST',
+            RestConfigurableProducer::REQUEST_BODY => ['hello' => 'world'],
+            RestConfigurableProducer::REQUEST_URI => $requestUri,
+            RestConfigurableProducer::DISPLAY_RESPONSE_ERROR => false,
+        ];
+
+        $options = [
+            RestConfigurableProtocol::OPTION_BASE_URI => $baseUri,
+            RestConfigurableProtocol::OPTION_ENCODING => RestConfigurableProtocol::ENCODING_JSON,
+            ConfigurableWebserviceProtocol::OPTION_CONNECT_TIMEOUT => 10,
+            ConfigurableWebserviceProtocol::OPTION_TIMEOUT => 10,
+            RestConfigurableProtocol::OPTION_HEADERS => [],
+            RestConfigurableProtocol::OPTION_AUTH => false,
+        ];
+        $context = [
+            'vars' => [],
+            'msg'  => new Message(new SerializableThing(), [], new Context()),
+            'exchange' => new Exchange(),
+        ];
+
+        //Decide to override the base_uri or not
+        if (false !== $overRideBaseUri) {
+            $actionParams['base_uri'] = $overRideBaseUri;
+        }
+
+        $response = $producer->executeStep(RestConfigurableProducer::STEP_REQUEST, $actionParams, $options, $context);
+        $this->assertTrue($response);
```

Comment:
Please provide a meaningful message as second argument here, as 'Failing asserting that false is true' is not much help.

Reply:
added